From f81615df0d427608287ece6f08e3ec2e030e84c5 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Wed, 19 Aug 2015 15:32:44 +0800
Subject: [PATCH 0243/5242] MLK-11385 dma: imx-sdma: enable clock before
 context restored

commit  97abc8d25ceb9f23f38b5e7b24c22b7d35374a53 from
https://source.codeaurora.org/external/imx/linux-imx.git

fix sdma driver resume back failed if SDMA clock disabled before suspend:

PM: noirq resume of devices complete after 0.802 msecs
imx-sdma 30bd0000.sdma: Timeout waiting for CH0 ready
imx-sdma 30bd0000.sdma: loaded firmware 4.1
imx-sdma 30bd0000.sdma: restore context error!
dpm_run_callback(): sdma_resume+0x0/0x1c8 returns -110
PM: Device 30bd0000.sdma failed to resume early: error -110

Signed-off-by: Robin Gong <b38343@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 55b58bc..d3fa138 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -2280,8 +2280,6 @@ static int sdma_resume(struct device *dev)
 
 	/* prepare priority for channel0 to start */
 	sdma_set_channel_priority(&sdma->channel[0], MXC_SDMA_DEFAULT_PRIORITY);
-	clk_disable(sdma->clk_ipg);
-	clk_disable(sdma->clk_ahb);
 
 	ret = sdma_get_firmware(sdma, sdma->fw_name);
 	if (ret) {
@@ -2295,6 +2293,9 @@ static int sdma_resume(struct device *dev)
 		return ret;
 	}
 
+	clk_disable(sdma->clk_ipg);
+	clk_disable(sdma->clk_ahb);
+
 	return 0;
 }
 #endif
-- 
1.7.9.5

